src/codegen.cpp

```diff
@@ -5219,11 +5219,50 @@ static enum ZigLLVM_AtomicRMWBinOp to_ZigLLVMAtomicRMWBinOp(AtomicRmwOp op, bool
     zig_unreachable();
 }
 
+static LLVMTypeRef get_atomic_abi_type(CodeGen *g, IrInstGen *instruction) {
```

Comment:
This is an ad-hoc version of `LLVMABISizeOfType`.
One more idea, since you're already converting values forth and back you may also want to tackle #4457 in a general way by simply bitcasting floats back to an appropriately-sized `iN` type to gently sidestep this limitation in some backends.

Reply:
>you may also want to tackle #4457 in a general way

That would work for `.Xchg` but we also support `.Add` and `.Sub` for floats.

Reply:
Oh lord, I assumed only `Xchg` was valid for `floats`. Carry on then.